tests/tests_listen.yml

```diff
@@ -0,0 +1,56 @@
+
+- name: Ensure that rsyslog configured as a server listens on the port 514
+  hosts: all
+  become: true
+
+  tasks:
+    - name: Deploy rsyslog config on the target host as a server
+      vars:
+        logging_role_path: "roles/linux-system-roles.logging"
+        logging_enabled: True
+        logging_purge_confs: True
+        rsyslog_default: False
+        rsyslog_capabilities: [ 'network', 'remote-files' ]
+        logging_outputs:
+          - name: files-output
+            type: files
+            logs_collections:
+              - name: basics
+
+      include_role:
+        name: linux-system-roles.logging
+
+    - include: set_rsyslog_variables.yml
+
+    # notify restart rsyslogd is executed at the end of this test task.
+    # thus we have to manually restart rsyslogd here.
```

Comment:
use meta: flush_handlers for that. Or put your validation steps into a post_tasks section - Ansible apparently inserts meta: flush_handlers automatically before this section. 
https://docs.ansible.com/ansible/latest/modules/meta_module.html

Reply:
Thanks, @pcahyna.  Fixed.